python/mpi_cmake_modules/clang_format.py

```diff
@@ -61,30 +48,48 @@ def load_clang_format_config():
             if "_clang-format" in filenames:
                 return yaml2oneline(path.join(dirpath, "_clang-format"))
             break
+    raise Exception("failed to find _clang-format file in "+str(mpi_cmake_modules.__path__))
 
-
-def test_valid_file(filename):
-    """ Test if the input file is a valid C/C++ file.
-    
-    Look at the extension of the file and return if it is a good one or not.
+def test_valid_file(filename,extensions):
+    """ Test if the input file exists and is of one of the provided extension.
 
     Args:
         filename: str `--` Path to the file to test.
+        extensions: iterable of accepted extensions (str)
     
     Returns:
-        True if the file ends with one of the following extension
-            (".h", ".c", ".hh", ".cc", ".hpp", ".cpp", ".hxx", ".cxx")
+        True if the file exits and ends with one of the extension.
         False otherwise.
     """
     if (path.isfile(filename) and any([filename.endswith(suffix) for suffix in
-                                       (".h", ".c", ".hh", ".cc", ".hpp",
-                                        ".cpp", ".hxx", ".cxx")])):
+                                       extensions])):
         return True
     else:
         return False
 
+def get_absolute_path(file_or_directory):
+    """
+    Check if file_or_directory is an existing file or directory,
+    if so, returns it. Otherwise, assumes it is a relative path, 
+    which it upgrades to an absolute path, this it returns. If the 
+    upgraded path does not correspond to an existing file or directory,
+    returns None
+
+    Args:
+          file_or_directory: str `--` a relative or an absolute path
 
-def list_of_files_to_format(files_or_directories):
+    Returns:
+          an absolute existing path (str) or None
+    """
+    if path.isfile(file_or_directory) or path.isdir(file_or_directory):
```

Comment:
This part is useless (I think because `os.path.abspath(os.path.abspath(X)) == os.path.abspath(X)`
Which means you can simply execute the `fixed_path = os.path.abspath(file_or_directory)` and then test if is ok.

Reply:
I was thinking the same, and I tried a version using the suggested approach, and it did not work.